backend/controllers/LocationsController.ts

```diff
@@ -93,26 +93,43 @@ const LocationsController = {
       return res.status(500).json({ error: err.message });
     }
   },
+  async updateLocationById(req: Request, res: Response) {
+    try {
+      const id = req.params.id;
+
+      const location = await Location.findByIdAndUpdate(
+        id,
+        {
+          $set: { ...req.body }
+        },
+        { new: true }
+      ).exec();
```

Comment:
before putting values into db you should check the presence of all required data

Reply:
in addition to this, you have to just put _id of the user in the "likes" or "dislikes" array of the "rating" object!

Reply:
We don't need to do this! Mongoose is already doing it for us.